lib/coercive.rb

```diff
@@ -152,15 +152,20 @@ def member(set)
 
   # Public DSL: Return a coerce function to coerce input to an Integer.
   # Used when declaring an attribute. See documentation for attr_coerce_fns.
-  def integer
+  def integer(min: nil, max: nil)
     ->(input) do
       fail Coercive::Error.new("float_not_permitted") if input.is_a?(Float)
 
-      begin
-        Integer(input)
-      rescue TypeError, ArgumentError
-        fail Coercive::Error.new("not_numeric")
-      end
+      input = begin
+                Integer(input)
+              rescue TypeError, ArgumentError
+                fail Coercive::Error.new("not_numeric")
+              end
+
+      fail Coercive::Error.new("too_small") if min && input < min
+      fail Coercive::Error.new("too_large") if max && input > max
```

Comment:
Putting on my pedantry hat for a moment, "too small" is not quite accurate if I were to give a very large negative number - "too low" is accurate both for positive numbers that are too small, and for negative numbers that are too far into the negative.

```suggestion
      fail Coercive::Error.new("too_low") if min && input < min
      fail Coercive::Error.new("too_high") if max && input > max
```

Reply:
Pedantry makes software better. I'll take it.